pkg/planner/cascades/rule/rule.go

```diff
@@ -54,22 +60,38 @@ func NewBaseRule(tp ruleType, pattern *pattern.Pattern) *BaseRule {
 	}
 }
 
+// ID returns the basic ordinal this rule enum type.
+func (r *BaseRule) ID() uint {
+	// impl it in your own rule.
+	return 0
+}
+
+// Promise value return the default promising value of 1.
+func (r *BaseRule) Promise() int {
+	return 1
+}
```

Comment:
What's the usage of this function? And it seems that this function is not registered at the interface.

Reply:
currently no that necessary, ideally, the physical rule should have a promise value of 2, and the logical rule should have a default 1, this could make the valid rules be sorted with promise, leading to physical-impl be scheduled ahead of logical rules, then derive a base-cost-line for pruning.
but for now, we only think about logical rules, so the promise value is always 1 as returned. just not classify them in the interface.